### DISCUSSION TITLE
Figure out user flow

### DISCUSSION THREAD

**Discussion by NobleTyler** [codetocloudinc/code-to-cloud-home#11] [General]
Requirement it meets: Design Consistency*

Purpose: User flows helps us determine the placement of navigation elements and the hierarchy of information, ensuring a consistent and intuitive experience for users (

Considerations:

1.     Where is the first thing that your user interacts with in this experience
2.     What the user would achieve at the end of the experience
3.     What happens in-between
4.     accessibility considerations

Requirement it meets: Cross-device combability

Purpose: Mapping user flows helps identify potential friction points on different devices/browsers, enabling adjustments to maintain usability.

Considerations:

-     Can we perform the same tasks seamlessly across multiple devices? Example: Ensure users can start an action (e.g., registering for an event, reading a blog, or purchasing a ticket) on one device and seamlessly continue or complete it on another without losing progress.
---

This was taken from https://github.com/orgs/codetocloudinc/projects/6/views/1?layout=board&pane=issue&itemId=88012064 and blocks our [Figma ](https://github.com/codetocloudinc/code-to-cloud-home/issues/7)designs. Once we figure out this Flow we can begin on designs!

**Comment by haywirebanana** [comment]
All those considerations are great to think about! 

For my previous projects, and what I can see with our live demo of Project Hoverboard, the separation between registered/non-registered users might be crucial in both requirements of Design Consistency and Cross-device compatibility. 

General information/pages such as a **landing page**, blog posts, and a team member page can be accessed easily by everyone. This would help in the consideration of our consideration of "What the user will achieve at the end of the experience" of general information. 

As for registered users, having that sign-in/sign-up option would help in the consideration of "performing the same tasks across multiple devices". As users interact with registered-only functionality such as buying tickets or posting/reacting to blog posts, we can save that information as they go through the pages, ensuring their progress is saved as they move from devices. 

One restriction I see with this is that if we continue with Astro, its static site nature may restrict these functional requirements. However, implementing external CMS to render new content (for blogs) and possibly firebase to handle user authentication could be solutions for dynamic content and partial hydration.

**Reply by NobleTyler** [reply]
Currently the signup/registration will be handled by the ticketing platform. As for the blog I'm currently assuming it'll be just @kevinevans1 hosting his articles on there I'm assuming without comments for the early stage. 

Astro luckily works with Firebase and other CMS, if we did need to add commenting/reacting. However given the social strategy of CodeToCloud I'm thinking most comments/reactions will happen on the socials their posted on rather than this site.

**Comment by victoriacheng15** [comment]
Users can view events, blogs, and general information about Code to Cloud without restrictions, ensuring their first interaction with the platform is straightforward and engaging. This design choice aligns with the goal of guiding users through the experience intuitively. By the end, users should have a clear understanding of the purpose of the platform and its offerings.

By having a sign-in/sign-up page, users can have their progress saved and then continue unfinished actions on other devices. For example, they could start purchasing a ticket on Device A and seamlessly continue the process on Device B. This ensures a consistent experience across devices, allowing users to perform tasks without losing progress.

**Reply by NobleTyler** [reply]
Good call with ensuring fist interaction is straightforward and engaging. As it stands the only registration users will do will likely be with the ticketing platform, which isn't handled by us.

So it should be relatively simple 👍

**Comment by NobleTyler** [comment]
@nataliey16 Did you get to talk to @kevinevans1 yet about what he wants for this?

Also do we have a Figma setup? I'm thinking some of us could try using the Figma to design the site if so. Once we do that we can get some tickets going based on the designs and start coding. Thanks :) and let me know if you need any help.

**Reply by nataliey16** [reply]
Kevin said to follow the same flow and format of the DevFest website with the Code to Cloud colours! I have set up a Figma and I will create an initial draft with a Style Tile template of the colours and fonts. Im planning to have the initial designs done before Monday so we can discuss/review/adjust as a team for next week!
